data/ui/import.blp

```diff
@@ -38,61 +38,70 @@ template $ImportWindow : Adw.Window {
-        }
 
-        Adw.ComboRow separator {
-          title: _("Decimal Separator");
-          subtitle: _("Character used as decimal separator");
-        }
+      Box {
+        orientation: vertical;
+        spacing: 10;
+        Adw.PreferencesGroup columns_group {
+          visible: false;
+          Adw.EntryRow columns_delimiter {
+            max-width-chars: 10;
+            title: _("Delimiter (\s+ for whitespace)");
+          }
 
-        Adw.ActionRow {
-          title: _("Column X");
-          subtitle: _("X-data column index");
-          SpinButton column_x {
-            valign: center;
-            numeric: true;
-            value: 0;
-            wrap: true;
-            adjustment: Adjustment {
-              step-increment: 1;
-              upper: 100;
-            };
+          Adw.ComboRow columns_separator {
+            title: _("Decimal Separator");
           }
-        }
 
-        Adw.ActionRow {
-          title: _("Column Y");
-          subtitle: _("Y-data column index");
-          SpinButton column_y {
-            valign: center;
-            numeric: true;
-            value: 0;
-            wrap: true;
-            adjustment: Adjustment {
-              step-increment: 1;
-              upper: 100;
-            };
+          Adw.ActionRow {
+            title: _("Column X");
+            subtitle: _("X-data column index");
+            SpinButton columns_column_x {
+              valign: center;
+              numeric: true;
+              value: 0;
+              wrap: true;
+              adjustment: Adjustment {
+                step-increment: 1;
+                upper: 100;
+              };
+            }
           }
-        }
 
-        Adw.ActionRow {
-          title: _("Skip Rows");
-          subtitle: _("Ignored row indices");
-          SpinButton skip_rows {
-            valign: center;
-            numeric: true;
-            value: 0;
-            wrap: true;
-            adjustment: Adjustment {
-              step-increment: 1;
-              upper: 999;
-            };
+          Adw.ActionRow {
+            title: _("Column Y");
+            subtitle: _("Y-data column index");
+            SpinButton columns_column_y {
+              valign: center;
+              numeric: true;
+              value: 0;
+              wrap: true;
+              adjustment: Adjustment {
+                step-increment: 1;
+                upper: 100;
+              };
+            }
           }
+
+          Adw.ActionRow {
+            title: _("Skip Rows");
+            subtitle: _("Ignored row indices");
+            SpinButton columns_skip_rows {
+              valign: center;
+              numeric: true;
+              value: 0;
+              wrap: true;
+              adjustment: Adjustment {
+                step-increment: 1;
+                upper: 999;
+              };
+            }
+          }
+        }
+
+        CheckButton save_values {
```

Comment:
We probably want to use a Switch here instead of a toggle.

Also, the label should be a bit more descriptive I think. Like "Remember the above options" or something similar

Reply:
I haven't given much thought to the actual UI, as I wanted to get the technical side correctly first.

I don't feel a switch would be the better choice here, as having a checkbox helps distinguishing it from actual parameters.

I'd change the UI side in a separate pr, which tackles a few other issues as well.

Reply:
I've done quite some testing on the technical side of this PR, and no issues there. Seems to work exactly as intended.

Regarding the toggle/checkbox, my personal preference definitely goes to a switch here. A toggle quite clearly symbolizes the "on/off" state, and it's in my view just as distinct as a checkbox (neither of these are part of the other settings either). 

Just for a third opinion, I asked for advice in the GNOME Design Matrix channel, presenting the following four options.

![image](https://github.com/Sjoerd1993/Graphs/assets/68477016/e9d5bedd-b22c-43e4-8eb4-7e219c8ae0be)